src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -86,24 +104,85 @@ def add_helm_repo(kube_config, kube_context):
-    query_parameters = {}
-    query_parameters['releaseTrain'] = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
-    header_parameters = {}
-    header_parameters['Authorization'] = "Bearer {}".format(str(token))
+    release_train = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
+    uri_parameters = ["releaseTrain={}".format(release_train)]
     try:
-        response = requests.post(get_chart_location_url, params=query_parameters, headers=header_parameters)
+        r = send_raw_request(cmd.cli_ctx, 'post', get_chart_location_url, uri_parameters=uri_parameters, resource='https://management.core.windows.net/')
     except Exception as e:
         telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
                                 summary='Error while fetching helm chart registry path')
         raise CLIError("Error while fetching helm chart registry path: " + str(e))
-    if response.status_code == 200:
-        return response.json().get('repositoryPath')
-    telemetry.set_exception(exception=str(response.json()), fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
-                            summary='Error while fetching helm chart registry path')
-    raise CLIError("Error while fetching helm chart registry path: {}".format(str(response.json())))
+    if r.content:
+        try:
+            return r.json().get('repositoryPath')
+        except Exception as e:
+            telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                    summary='Error while fetching helm chart registry path')
+            raise CLIError("Error while fetching helm chart registry path from JSON response: " + str(e))
+    else:
+        telemetry.set_exception(exception='No content in response', fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                summary='No content in acr path response')
+        raise CLIError("No content was found in helm registry path response.")
+
+
+def arm_exception_handler(ex, fault_type, summary, return_if_not_found=False):
+    if isinstance(ex, AuthenticationError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Authentication error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, TokenExpiredError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Token expiration error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, HttpOperationError):
+        status_code = ex.response.status_code
+        if status_code == 404 and return_if_not_found:
+            return
```

Comment:
just curious, why just return when 404?

Reply:
We are here just checking if the resource already exists before creating it. Hence we want to just return without throwing error